### PR TITLE
add: `mission-center`

### PR DESCRIPTION
**Pull request by AozoraDev**
i don't want to use flatpak anymore.
so here's the native one.

### CODE REVIEW

**Comment by AozoraDev** [comment]
Tested on Debian Trixie.
For Ubuntu, idk.

**Comment by AozoraDev** [comment]
related issue about failed tests
[pacstall#1233](https://github.com/pacstall/pacstall/issues/1233)

**Comment by Xdavius** [comment]
> related issue about failed tests [pacstall#1233](https://github.com/pacstall/pacstall/issues/1233)

add : 

external_connection=true 

https://github.com/pacstall/pacstall/wiki/Pacscript-101
external_connection=true # Allow internet access during compilation process

File: packages/mission-center/mission-center.pacscript [hunk 1]
```diff
@@ -0,0 +1,47 @@
+pkgname="mission-center"
+pkgver="0.5.2"
+pkgrel="1"
```
Comment (vigress8) [inline packages/mission-center/mission-center.pacscript:3]:
```suggestion
```
`pkgrel` is optional in pacscripts, it will be 1 by default anyway

Comment (AozoraDev) [reply packages/mission-center/mission-center.pacscript]: i didn't remember i add that.
but eh, sure.

**Comment by AozoraDev** [comment]
aight. for now, i'll just make it incompatible with stable distro until i can make the patch for it.

**Comment by AozoraDev** [comment]
@vigress8 
Guess everything's good now.
I'll see if i can support stable distro in the future.

Also the rest failed tests was... internet connection problem(?)

**Review by vigress8 — APPROVED** [review]
For the future:
1) Run `pre-commit install` in your clone of the repo, to enable the pre-commit hooks
2) Squash your commits, just for cleanliness

**Comment by AozoraDev** [comment]
> For the future:
> 
> 1. Run `pre-commit install` in your clone of the repo, to enable the pre-commit hooks
> 2. Squash your commits, just for cleanliness

image

File: packages/mission-center/mission-center.pacscript [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  "libudev-dev"
+  "libgbm-dev"
+  "libpango1.0-dev"
+  "libgdk-pixbuf-2.0-dev"
+  "libgtk-4-dev"
+  "libdbus-1-dev"
+  "libadwaita-1-dev"
+)
+checkdepends=("appstream-util")
+source=("${url}/-/archive/v${pkgver}/${pkgname}-v${pkgver}.tar.gz")
```
Comment (Elsie19) [inline packages/mission-center/mission-center.pacscript:29]: I don't like that url is a variable here. It's not "obfuscated" but it doesn't feel right.

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: packages/mission-center/mission-center.pacscript (hunk 2 above)
Comment (AozoraDev) [reply packages/mission-center/mission-center.pacscript:29]: Uhm.... it looks fine to me....
But eh, since you worry about that....

Comment (AozoraDev) [reply packages/mission-center/mission-center.pacscript]: [#634264c](https://github.com/pacstall/pacstall-programs/pull/6383/commits/634264cf7425bb6a37ff0adf483717e7b9031e5a)

Happy?

File: packages/mission-center/mission-center.pacscript [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  "libudev-dev"
+  "libgbm-dev"
+  "libpango1.0-dev"
+  "libgdk-pixbuf-2.0-dev"
+  "libgtk-4-dev"
+  "libdbus-1-dev"
+  "libadwaita-1-dev"
+)
+checkdepends=("appstream-util")
+source=("https://gitlab.com/mission-center-devs/mission-center/-/archive/v${pkgver}/mission-center-v${pkgver}.tar.gz")
```
Comment (AozoraDev) [inline packages/mission-center/mission-center.pacscript:29]: It looks not "obfuscated" now.

**Comment by Elsie19** [comment]
Extatic 😁

-------- Original Message --------

**Comment by oklopfer** [comment]
this shouldn’t have been changed. @Elsie19 was being stupid. This is standard practice. It doesn’t matter if she “doesn’t like it.”

**Comment by Xdavius** [comment]
> Also lemme see if i can fix for Debian Stable...

I need to pacscript rust-bin and test if it build and work with new version of rust.

In this case, it will be possible to add something like :

if [[ ${DISTRO} = "bookworm" ]]; then
  pacdeps=(rust-bin)
else
  makedepends+=("rustc" "cargo")
fi

**Comment by AozoraDev** [comment]
@Xdavius 
Nah i was thinking to just use old version of mission-center since stable distros using old packages too...

**Comment by Xdavius** [comment]
> @Xdavius Nah i was thinking to just create another pacscript of old version of mission-center since stable distros using old packages too...

hummm...

I have tested with latest 1.80.1 Rust version but still an error : 

```
[32/37] Generating resources/blueprints with a custom command
FAILED: resources 
/usr/bin/blueprint-compiler batch-compile resources/. ../resources ../resources/ui/apps_page/column_header.blp ../resources/ui/apps_page/list_item.blp ../resources/ui/apps_page/page.blp ../resources/ui/apps_page/pid_column.blp ../resources/ui/apps_page/stat_column.blp ../resources/ui/performance_page/cpu.blp ../resources/ui/performance_page/cpu_details.blp ../resources/ui/performance_page/disk.blp ../resources/ui/performance_page/disk_details.blp ../resources/ui/performance_page/gpu.blp ../resources/ui/performance_page/gpu_details.blp ../resources/ui/performance_page/memory.blp ../resources/ui/performance_page/memory_details.blp ../resources/ui/performance_page/network.blp ../resources/ui/performance_page/network_details.blp ../resources/ui/performance_page/page.blp ../resources/ui/performance_page/summary_graph.blp ../resources/ui/preferences/page.blp ../resources/ui/preferences/window.blp ../resources/ui/services_page/page.blp ../resources/ui/services_page/context_menu_button.blp ../resources/ui/services_page/details_dialog.blp ../resources/ui/window.blp ../resources/ui/gtk/help-overlay.blp
error: Could not determine what kind of syntax is meant here
at ../resources/ui/apps_page/column_header.blp line 23 column 10:
  23 |template $ColumnHeader: Box {
     |         ^

[34/37] Generating data/hwdb/hwdb with a custom command
ninja: build stopped: subcommand failed.
```

**Comment by AozoraDev** [comment]
> > @Xdavius Nah i was thinking to just create another pacscript of old version of mission-center since stable distros using old packages too...
> 
> hummm...
> 
> I have tested with latest 1.80.1 Rust version but still an error :
> 
> ```
> [32/37] Generating resources/blueprints with a custom command
> FAILED: resources 
> /usr/bin/blueprint-compiler batch-compile resources/. ../resources ../resources/ui/apps_page/column_header.blp ../resources/ui/apps_page/list_item.blp ../resources/ui/apps_page/page.blp ../resources/ui/apps_page/pid_column.blp ../resources/ui/apps_page/stat_column.blp ../resources/ui/performance_page/cpu.blp ../resources/ui/performance_page/cpu_details.blp ../resources/ui/performance_page/disk.blp ../resources/ui/performance_page/disk_details.blp ../resources/ui/performance_page/gpu.blp ../resources/ui/performance_page/gpu_details.blp ../resources/ui/performance_page/memory.blp ../resources/ui/performance_page/memory_details.blp ../resources/ui/performance_page/network.blp ../resources/ui/performance_page/network_details.blp ../resources/ui/performance_page/page.blp ../resources/ui/performance_page/summary_graph.blp ../resources/ui/preferences/page.blp ../resources/ui/preferences/window.blp ../resources/ui/services_page/page.blp ../resources/ui/services_page/context_menu_button.blp ../resources/ui/services_page/details_dialog.blp ../resources/ui/window.blp ../resources/ui/gtk/help-overlay.blp
> error: Could not determine what kind of syntax is meant here
> at ../resources/ui/apps_page/column_header.blp line 23 column 10:
>   23 |template $ColumnHeader: Box {
>      |         ^
> 
> [34/37] Generating data/hwdb/hwdb with a custom command
> ninja: build stopped: subcommand failed.
> ```

Skill issue

lol jk. i have no idea too. i can't read rust.

**Comment by Xdavius** [comment]
> Skill issue
In fact it was ^^ 
> lol jk. i have no idea too. i can't read rust.

It's blueprint-compiler error... 
and building blueprint compiler gives weird with missing sphinx theme.

It's dead for Debian 12

**Comment by AozoraDev** [comment]
Yep it takes too long. Gotta maintain it by myself then.

**Pull request closed without merging**